### ISSUE TITLE
Composer 2 support

### ISSUE THREAD

**Issue by vielhuber** [humbug/php-scoper#452]
## Environment

| Question    | Answer
| ------------| ---------------
| Box version | 0.13.10
| PHP version | 7.4.9
| Platform with version | Ubuntu

## Scenario 1

```sh
composer self-update --1
composer --version # 1.10.19
```

**Behaviour: php-scoper prefixes all classes as desired.**

## Scenario 2

```sh
composer self-update --2
composer --version # 2.0
```

**Behaviour: php-scoper does not prefix any classes (can be seen in autoload_real.php).**

## Question

Whats the current status of php-scopers support of Composer 2?

Is there any support for Composer 2 planned?

**Comment by theofidry** [comment]
Note that regardless of the Composer version, you need to dump the autoloader again.

Is the rest of the code not prefixed at all?

**Comment by vielhuber** [comment]
> Note that regardless of the Composer version, you need to dump the autoloader again.

Yes, that's what's happening.

> Is the rest of the code not prefixed at all?

Inside the php files the classes are prefixed, but as mentionned in the autoload_real.php they are not.

**Comment by villfa** [comment]
I fail to reproduce the problem.

@vielhuber may you share the exact steps to reproduce it with your `scoper.inc.php` file? A repro repository would be ideal.

**Comment by vielhuber** [comment]
### Test files

```composer.json```
{
    "name": "vielhuber/test",
    "description": "Test.",
    "license": "MIT",
    "autoload": {
        "psr-4": {
            "vielhuber\\Test\\": "src/"
        }
    },
    "require": {
        "vielhuber/dbhelper": "*"
    }
}
```

**scoper.inc.php**
```php
<?php
declare(strict_types=1);
return [
    'prefix' => 'ScopedGtbabel',
    'whitelist-global-functions' => false,
    'whitelist-global-constants' => false,
    'whitelist-global-classes' => false,
    'patchers' => []
];
```

**src/Test.php**

```php
namespace vielhuber\test;
use vielhuber\dbhelper\dbhelper;
class Test
{
    function __construct() {
        new dbhelper();
    }
}
```

### composer 1

rm -rf vendor/; rm composer.lock; composer self-update --1 && composer install
rm -rf build/ && php ./php-scoper.phar add-prefix --config scoper.inc.php
cd build/ && composer dump-autoload && cd ..
cat build/vendor/composer/autoload_static.php

```php
/* ... */
public static $prefixDirsPsr4 = array (
        'ScopedGtbabel\\vielhuber\\dbhelper\\' =>
        array (
            0 => __DIR__ . '/..' . '/vielhuber/dbhelper/src',
        ),
/* ... */
```

### composer 2

rm -rf vendor/; rm composer.lock; composer self-update --2 && composer install
rm -rf build/ && php ./php-scoper.phar add-prefix --config scoper.inc.php
cd build/ && composer dump-autoload && cd ..
cat build/vendor/composer/autoload_static.php

```php
/* ... */
public static $prefixDirsPsr4 = array (
        'vielhuber\\dbhelper\\' =>
        array (
            0 => __DIR__ . '/..' . '/vielhuber/dbhelper/src',
        ),
/* ... */
```

**Comment by avdv** [comment]
Hi.

I basically hit the same problem. Trying to reproduce this, it seems this problem is fixed with php-scoper 0.14.0.

Note, you don't have to dump the files, just look into the `vendor/composer/installed.json` file and ensure that `psr-4` entries are properly prefixed (when dumping, composer just looks at this file and creates the autoload information AFAIICS).

I am using PHP 7.4.15, composer 2.0.11, php scoper 0.14. For the code from @vielhuber:
```console
$ composer install
$ php-scoper add-prefix -c scoper.inc.php 


    ____  __  ______     _____
   / __ \/ / / / __ \   / ___/_________  ____  ___  _____
  / /_/ / /_/ / /_/ /   \__ \/ ___/ __ \/ __ \/ _ \/ ___/
 / ____/ __  / ____/   ___/ / /__/ /_/ / /_/ /  __/ /
/_/   /_/ /_/_/       /____/\___/\____/ .___/\___/_/
                                     /_/

PHP Scoper version 0.14.0 2020-12-21 14:36:07 UTC

 35/35 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] Successfully prefixed 35 files.                                                                                   
                                                                                                                        

 // Memory usage: 18.85MB (peak: 18.94MB), time: 0.31s                                                                  
$ grep psr-4 -A 2 build/vendor/composer/installed.json 
grep psr-4 -A 2 build/vendor/composer/installed.json 
                "psr-4": {
                    "ScopedGtbabel\\vielhuber\\dbhelper\\": "src\/",
                    "ScopedGtbabel\\Tests\\": "tests\/"
```

I could reproduce the problem with scoper 0.13.2:
```console
$ composer install
$ php-scoper add-prefix
$ php-scoper add-prefix -c scoper.inc.php 


    ____  __  ______     _____                           
   / __ \/ / / / __ \   / ___/_________  ____  ___  _____
  / /_/ / /_/ / /_/ /   \__ \/ ___/ __ \/ __ \/ _ \/ ___/
 / ____/ __  / ____/   ___/ / /__/ /_/ / /_/ /  __/ /    
/_/   /_/ /_/_/       /____/\___/\____/ .___/\___/_/     
                                     /_/

PHP Scoper version 0.13.2 2020-06-14 08:19:08 UTC

 35/35 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] Successfully prefixed 35 files.                                                                                   
                                                                                                                        

 // Memory usage: 11.10MB (peak: 17.85MB), time: 0.30s                                                                  
$ grep psr-4 -A 2 build/vendor/composer/installed.json 
                "psr-4": {
                    "vielhuber\\dbhelper\\": "src/",
                    "Tests\\": "tests/"
```

**Comment by theofidry** [comment]
Fixed via 0.16.
